src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -35,8 +36,9 @@ internal sealed class HttpLoggingMiddleware
         /// </summary>
         /// <param name="next"></param>
         /// <param name="options"></param>
-        /// <param name="logger"></param>
-        public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOptions> options, ILogger<HttpLoggingMiddleware> logger)
+        /// <param name="loggerFactory"></param>
+        /// <param name="filterOptions"></param>
+        public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOptions> options, ILoggerFactory loggerFactory, IOptionsMonitor<LoggerFilterOptions> filterOptions)
```

Comment:
Is there going to be a way to hook this middleware directly with the W3CLogger? Maybe W3CLogger should be made public so it's easy to bypass DI for this if you want to. It'd be an easy way to avoid sending logs to other providers if you don't want to.

Reply:
Seems like a good idea - Is there even any need to make the LoggerProvider public? Seems to me like we should just expose the extension methods (`AddW3CLogger`).

Reply:
Right now I'm thinking we should make the entire logger part of this PR internal and hook it up with the middleware automatically. I can see the point of having this middleware also work with normal loggers, but I do not see the point of the W3CLogger receiving logs from anything but this middleware.

Reply:
> This middleware could just new up the W3CLogger and people could opt in to logging the W3C logs using the ILoggerFactory instead of or in addition to the W3CLogger.

It isn't obvious to me what that would look point from the user standpoint, could you give an example?

Reply:
It's a big change, but we can schedule a meeting. `Microsoft.Extensions.Logging.W3C` would cease to exist. All of it would have to be folded into `Microsoft.AspNetCore.HttpLogging`. Almost all of it would be made internal except for `W3CLoggerOptions` which would be added to `HttpLoggingOptions` as a property or something.

Reply:
That part makes sense, I'm just not clear on how the user controls whether or not W3C logging is on. In any case I can schedule something quick for tomorrow.

Reply:
> I'm just not clear on how the user controls whether or not W3C logging is on.

Why can't this all be controlled by HttpLoggingOptions?

Reply:
> Why can't this all be controlled by HttpLoggingOptions?

Ah, it could. Good idea

Reply:
Done